app/src/main/java/ai/elimu/appstore/BaseApplication.java

```diff
@@ -73,6 +78,15 @@ protected void log(int priority, String tag, String message, Throwable throwable
         daoSession = new CustomDaoMaster(db).newSession();
 
         VersionHelper.updateAppVersion(getApplicationContext());
+
+        /**
+         * Check and delete existing APKs to avoid manually deleting corrupt files before upgrading to version 2.0.9
+         * of the Appstore application
+         */
+        int newVersionCode = VersionHelper.getAppVersionCode(context);
+        if (newVersionCode < 2000009) {
```

Comment:
Shouldn't this be `if (oldVersionCode < 2000009) {`?

Reply:
Actually it should be `if (oldVersionCode < 2000010) {`.
However, I'm thinking again if we should really put the delete APKs logic here.
If the check is here, when user's app is of version < 2000010, and he already downloaded some apps, after closing and opening app again, those APKs will be removed without user having to upgrade to latest app version.
Is that what we expect?

How about placing the deletion logic in database's onUpgrade?